x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/components/step_define_rule/schema.tsx

```diff
@@ -144,46 +142,8 @@ export const schema: FormSchema<DefineStepRule> = {
     ),
     validations: [],
   },
-  anomalyThreshold: {
-    label: i18n.translate(
-      'xpack.securitySolution.detectionEngine.createRule.stepDefineRule.fieldAnomalyThresholdLabel',
-      {
-        defaultMessage: 'Anomaly score threshold',
-      }
-    ),
-    validations: [],
-  },
-  machineLearningJobId: {
-    label: i18n.translate(
-      'xpack.securitySolution.detectionEngine.createRule.stepDefineRule.fieldMachineLearningJobIdLabel',
-      {
-        defaultMessage: 'Machine Learning job',
-      }
-    ),
-    validations: [
-      {
-        validator: (
-          ...args: Parameters<ValidationFunc>
-        ): ReturnType<ValidationFunc<{}, ERROR_CODE>> | undefined => {
-          const [{ formData }] = args;
-          const needsValidation = isMlRule(formData.ruleType);
-
-          if (!needsValidation) {
-            return;
-          }
-
-          return fieldValidators.emptyField(
-            i18n.translate(
-              'xpack.securitySolution.detectionEngine.createRule.stepDefineRule.machineLearningJobIdRequired',
-              {
-                defaultMessage: 'A Machine Learning job is required.',
-              }
-            )
-          )(...args);
-        },
-      },
-    ],
-  },
+  anomalyThreshold: {},
```

Comment:
Are these keys still necessary if we're now defining all this behavior within individual field `config` ?

Reply:
It's still necessary for now. Schema keys are [used](https://github.com/nikitaindik/kibana/blob/0933e39d74825294cc68133aa75b729dce1f6825/x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/components/description_step/index.tsx#L101-L102) to display field components in step preview on the Rule Creation page.